src/tools/tidy/src/x.rs

```diff
@@ -0,0 +1,19 @@
+use std::process::Command;
+
+pub fn check(_bad: &mut bool) {
+    let result = Command::new("x")
+        .arg("--version")
+        .output();
+    let output = match result {
+        Ok(output) => output,
+        Err(_e) => todo!(),
+    };
+
+    if output.status.success() {
+        let version = String::from_utf8_lossy(&output.stdout);
+        assert_eq!("0.1.0", version.trim_end());
+    }
+    // FIXME: throw some kind of tidy error when the version of x isn't
+    // greater than or equal to the version we'd expect.
+    //tidy_error!(bad, "Current version of x is {version} Consider updating to the newer version of x by running `cargo install --path src/tools/x`")
```

Comment:
ok, so a few things:
- this should do a >= comparison, not just strictly equal, which means you need to pull in https://docs.rs/semver/latest/semver/ so you can compare versions
- bootstrap already has a version number and it's not the same as `x`'s version number https://github.com/rust-lang/rust/blob/70fe5f08fffd16dc20506f7d140e47b074f77964/src/bootstrap/lib.rs#L194 I am not sure how to reconcile this; maybe we can make the flag different and only interpreted by the `x` binary, not bootstrap itself? something like `--x-shim-version`?
- this should use `tidy_error` instead of `assert_eq` like you've already noticed; it can be something simple like `if version < "0.1.0" { tidy_error!(...) }`